File: examples/undocumented/ruby_modular/features_read_svmlight_format_modular.rb
require 'modshogun'
require 'pp'
parameter_list=[['../data/train_sparsereal.light']]

def features_read_svmlight_format_modular(fname)

	f=Modshogun::SparseRealFeatures.new
	lab=f.load_svmlight_file(Modshogun::LibSVMFile.new(fname))
	f.write_svmlight_file(Modshogun::LibSVMFile.new('testwrite.light', 'w'), lab)

end

if __FILE__ == $0
	puts 'Reading SVMLIGHT format'
	pp features_read_svmlight_format_modular(*parameter_list[0])
end

File: examples/undocumented/python_modular/features_read_svmlight_format_modular.py
#!/usr/bin/env python
parameter_list=[['../data/train_sparsereal.light']]

def features_read_svmlight_format_modular (fname):
	import os
	from modshogun import SparseRealFeatures
	from modshogun import LibSVMFile

	f=SparseRealFeatures()
	lab=f.load_svmlight_file(LibSVMFile(fname))
	f.write_svmlight_file(LibSVMFile('testwrite.light', 'w'), lab)
#	os.unlink('testwrite.light')

if __name__=='__main__':
	print('Reading SVMLIGHT format')
	features_read_svmlight_format_modular(*parameter_list[0])
